Lib/fontTools/feaLib/builder.py

```diff
@@ -1384,18 +1373,18 @@ def getAlternateGlyphs(self):
     def find_chainable_single_subst(self, glyphs):
         """Helper for add_single_subst_chained_()"""
         res = None
-        for _, _, _, substitutions in self.substitutions[::-1]:
-            if substitutions == self.SUBTABLE_BREAK_:
+        for _, _, _, rules in self.rules[::-1]:
+            if len(rules) > 0 and rules[0] == self.SUBTABLE_BREAK_:
```

Comment:
same here.. Did we break this subtable break logic when we switched to a list of lookups?

Reply:
No, nothing to do with that. It's just that we call the "lookup list" attribute `.rules` in the  positioning lookup builder and `.substitutions` in the substitution lookup builder. This makes it harder to share code in a superclass. So I renamed `.substitutions` to `.rules`.

Reply:
Sorry, that's something different.

Reply:
But something similar! Previously, when we wanted a subtable break in a *pos* rule, we did this:

        self.rules.append((self.SUBTABLE_BREAK_, self.SUBTABLE_BREAK_,
                           self.SUBTABLE_BREAK_, [self.SUBTABLE_BREAK_]))

and when we had a subtable break in a *sub* rule we did this:
        self.rules.append((self.SUBTABLE_BREAK_, self.SUBTABLE_BREAK_,
                           self.SUBTABLE_BREAK_, self.SUBTABLE_BREAK_))

There was no reason for them to be different, so I made them the same.

Reply:
i'm a bit confused.. The snippet above looks weird to me (with the last item of the tuple being a 1-item list). Were we really doing that?

Reply:
https://github.com/fonttools/fonttools/blob/6b4430a8a39774851b707f10d6b403017451893c/Lib/fontTools/feaLib/builder.py#L1326

Reply:
Happy to go the other way and make the old-*sub* style the default, but I thought it didn't really matter and we expect the lookups list to be a list anyway.

Reply:
i don't see immediately see why it has to be like that..

But my question was about this change (ignore the `substituions` being renamed to `rules`):

```diff
- if substitutions == self.SUBTABLE_BREAK_:
+ if len(rules) > 0 and rules[0] == self.SUBTABLE_BREAK_:
```

that's very different, and I am wondering where that comes from

Reply:
This is what I'm talking about. When we had a manual subtable break, ChainContextPosBuilder would push `[self.SUBTABLE_BREAK_]` onto the lookup list, but ChainContextSubstBuilder would push `self.SUBTABLE_BREAK_` onto the lookup list. I fixed this inconsistency (and refactored `add_subtable_break` to a superclass), by making both builders push `[self.SUBTABLE_BREAK_]`.

That meant that code in ChainContextSubstBuilder  which expected to test against `self.SUBTABLE_BREAK_` needed to be updated.